staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -111,35 +112,29 @@ func WithAudit(handler http.Handler, requestContextMapper request.RequestContext
 // - context with audit event attached to it
 // - created audit event
 // - error if anything bad happened
-func createAuditEventAndAttachToContext(requestContextMapper request.RequestContextMapper, req *http.Request, policy policy.Checker) (request.Context, *auditinternal.Event, []auditinternal.Stage, error) {
-	ctx, ok := requestContextMapper.Get(req)
-	if !ok {
-		return nil, nil, nil, fmt.Errorf("no context found for request")
-	}
+func createAuditEventAndAttachToContext(req *http.Request, policy policy.Checker) (*http.Request, *auditinternal.Event, []auditinternal.Stage, error) {
```

Comment:
Can we not return a *http.Request?

Reply:
If you want to modify the context, you have to return the result of req.WithContext for use by the rest of the handler chain

Reply:
the input req is modified, they are equal

Reply:
Sorry， I just notified 
```
// WithContext returns a shallow copy of r with its context changed
// to ctx. The provided ctx must be non-nil.
func (r *Request) WithContext(ctx context.Context) *Request {
	if ctx == nil {
		panic("nil context")
	}
	r2 := new(Request)
	*r2 = *r
	r2.ctx = ctx

	// Deep copy the URL because it isn't
	// a map and the URL is mutable by users
	// of WithContext.
	if r.URL != nil {
		r2URL := new(url.URL)
		*r2URL = *r.URL
		r2.URL = r2URL
	}

	return r2
}
```

Reply:
actually a new address, a small trap

Reply:
Yeah, not the same object